api/src/main/java/org/bahmni/module/communication/api/impl/CommunicationServiceImpl.java

```diff
@@ -0,0 +1,148 @@
+package org.bahmni.module.communication.api.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.bahmni.module.communication.api.CommunicationService;
+import org.bahmni.module.communication.model.MailContent;
+import org.openmrs.api.context.Context;
+import org.openmrs.util.OpenmrsUtil;
+
+import javax.mail.Session;
+import javax.mail.Address;
+import javax.mail.Multipart;
+import javax.mail.Transport;
+import javax.mail.Authenticator;
+import javax.mail.PasswordAuthentication;
+import javax.mail.internet.*;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Properties;
+
+public class CommunicationServiceImpl implements CommunicationService {
+
+    private static final String EMAIL_PROPERTIES_FILENAME = "mail-config.properties";
+    private final Log log = LogFactory.getLog(this.getClass());
+    private volatile Session session = null;
+
+    @Override
+    public void sendEmail(MailContent mailContent) {
+        try {
```

Comment:
We have few limitations with **Context.getMessageService()** because of which we were not able to proceed. one of it is the email properties should be configured from global properties. And couple of other limitations we could use openmrs messageService

Reply:
@binduak latest openmrs core 2.5.x reads username and password in the following order 

1. JVM Args
2. OpenMRS Runtime Props
3. Global Props

Reference: https://github.com/openmrs/openmrs-core/blob/2.5.10/api/src/main/java/org/openmrs/api/context/Context.java#L643

https://github.com/openmrs/openmrs-core/blob/2.5.10/api/src/main/java/org/openmrs/util/ConfigUtil.java#L76

cc: @angshu @kavitha-sundararajan

Reply:
@sivareddytw session is not persisting the OpenMRS runtime props when Rohit has tried to use Context.getMessageService().